core/src/main/scala/com/avast/grpc/jsonbridge/ReflectionGrpcJsonBridge.scala

```diff
@@ -0,0 +1,151 @@
+package com.avast.grpc.jsonbridge
+
+import cats.effect.Async
+import cats.syntax.all._
+import com.avast.grpc.jsonbridge.GrpcJsonBridge.GrpcMethodName
+import com.google.common.util.concurrent.{FutureCallback, Futures, ListenableFuture}
+import com.google.protobuf.util.JsonFormat
+import com.google.protobuf.{Message, MessageOrBuilder}
+import com.typesafe.scalalogging.StrictLogging
+import io.grpc.MethodDescriptor.{MethodType, PrototypeMarshaller}
+import io.grpc._
+import io.grpc.inprocess.{InProcessChannelBuilder, InProcessServerBuilder}
+import monix.eval.Task
+import monix.execution.Scheduler
+
+import scala.collection.JavaConverters._
+import scala.concurrent.{Future, Promise}
+import scala.language.higherKinds
+import scala.util.control.NonFatal
+
+class ReflectionGrpcJsonBridge[F[_]](services: ServerServiceDefinition*)(implicit val scheduler: Scheduler, protected val F: Async[F])
```

Comment:
Thinking about it further, it should probably return `cats.effect.Resource` or at least `Task` because it creates some channels inside and it may take a while and definitely should be closed afterwards :thinking:

Reply:
And, don't forget about `BindableService` which is usually what user has in hands :wink:

Reply:
This is regular class (not case class) so I would like to preserve the usage of `new` keyword while constructing a new instance.

I've removed `Scheduler` from the public API.

`Async[F]` is `protected` in current implementation. I've changed it to public now.

It's OK to create a bridge for empty list of services (e.g. the main endpoint then returns empty response) so I want to force the users to use non-empty list of services.

Reply:
`BindableService` is actually easily convertible to `ServerServiceDefinition` (using `bindService()` method). Also the `BindableService` is used just for the _core_ implementation - if you apply any `ServerInterceptor` then you must work with `ServerServiceDefinition`.